base-notebook/Dockerfile

```diff
@@ -60,14 +60,14 @@ RUN apt-get update --yes && \
 # Configure environment
 ENV CONDA_DIR=/opt/conda \
     SHELL=/bin/bash \
-    NB_USER=$NB_USER \
-    NB_UID=$NB_UID \
-    NB_GID=$NB_GID \
+    NB_USER="${NB_USER}" \
+    NB_UID=${NB_UID} \
+    NB_GID=${NB_GID} \
```

Comment:
In a way, this difference between quotes or not on NB_USER vs NB_UID is not making sense, as both will be understood as strings in the end as they are set as environment variables can only be string values. I'm not sure what happens if someone sets NB_UID to "123 HELLO", would quote have mattered then?

I'd be fine leaving this as it is or adding quotes here as well.

Reply:
I suggest we don't think about this further and just go with the PR as it is. No change requested!

Reply:
> In a way, this difference between quotes or not on NB_USER vs NB_UID is not making sense

There is actually a difference.
NB_USER - is a username, it can be sth like "my best username ever". It can be with spaces so quoting it makes sense.
While NB_UID - is a Linux uid, which is numerical value. Quoting doesn't help (but doesn't make it worse).

```
# usermod -u 100 some_user
usermod: user 'some_user' does not exist
# usermod -u 100text some_user
usermod: invalid user ID '100text'
```

Reply:
I was thinking that NB_UID will hold a numerical value we assume, but it is a string representing a numerical value whenever we hold it in an environment variables, so in that sense, it is the same as NB_NAME even though we expect NB_UID to only contain numbers and no spaces etc.

Reply:
Yes, but, we can safely assume that NB_UID will have no spaces, while NB_USER easily might.
I don't like too excessive quoting where it won't help, because it might make reading a bit more difficult.
That's why I quoted only partially based on context and the meaning of variable.

Reply:
I agree!